api/agentpb/agent.proto

```diff
@@ -534,6 +534,25 @@ message JobProgress {
   }
 }
 
+// GetVersionRequest represents a request for version retrieving of different software.
+message GetVersionRequest {
+  // LocalMySQL is used for local MySQL version retrieving using `mysql --version`.
+  message LocalMySQL {}
+
+  // XtraBackup is used for xtrabackup version retrieving.
+  message XtraBackup {}
```

Comment:
Can we use enums instead of empty messages? Or maybe we can just return all software versions that we have at once? I mean:
```
message GetVersionsRequest {}
message GetVersionsResponse {
     string mysql_version = 1;
     xtrabackup_version = 2;
     ...
}
```

Reply:
Or we can pass a list of software that we need and get an array of versions as result.

Reply:
Yeah, enums could work in my case, but later we probably want to add a mongodb, and I suppose it will require some credentials, so the message suits better. Not sure we can return all the software versions at once, because some hosts could have MySQL only installed, others mongodb etc.

Reply:
Why mongodb will require credentials? The version check for it will differ from `mongod --version`?

My point that typically we need versions of some set of software (mongod + pbm, MySQL + xtrabackup + xtracloud, etc), so it's better to grab it In one request.


>  Not sure we can return all the software versions at once, because some hosts could have MySQL only installed, others mongodb etc.
Why it's the problem? If agent has only MySQL then it will return only MySQL version.

Reply:
As per our call in case of logical backup, we need to pass DB creds, so no better solution at the moment.